src/core/Entry.cpp

```diff
@@ -670,6 +672,102 @@ void Entry::updateModifiedSinceBegin()
+        const QString found = placeholderRegEx.cap(1);
+        result.replace(found, resolvePlaceholderRecursive(found, maxDepth - 1));
+        pos += placeholderRegEx.matchedLength();
+    }
+
+    if (result != str) {
+        result = resolveMultiplePlaceholdersRecursive(result, maxDepth - 1);
+    }
+
+    return result;
+}
+
+QString Entry::resolvePlaceholderRecursive(const QString &placeholder, int maxDepth) const
+{
+    const PlaceholderType placeholderType = this->placeholderType(placeholder);
+    switch (placeholderType) {
+    case PlaceholderType::NotPlaceholder:
+        return placeholder;
+    case PlaceholderType::Unknown:
+        qWarning("Can't resolve placeholder %s for entry with uuid %s", qPrintable(placeholder),
+                 qPrintable(uuid().toHex()));
+        return placeholder;
+    case PlaceholderType::Title:
+        return title();
+    case PlaceholderType::UserName:
+        return username();
+    case PlaceholderType::Password:
+        return password();
+    case PlaceholderType::Notes:
+        return notes();
+    case PlaceholderType::Totp:
+        return totp();
+    case PlaceholderType::Url:
+        return url();
+    case PlaceholderType::UrlWithoutScheme:
+    case PlaceholderType::UrlScheme:
+    case PlaceholderType::UrlHost:
+    case PlaceholderType::UrlPort:
+    case PlaceholderType::UrlPath:
+    case PlaceholderType::UrlQuery:
+    case PlaceholderType::UrlFragment:
+    case PlaceholderType::UrlUserInfo:
+    case PlaceholderType::UrlUserName:
+    case PlaceholderType::UrlPassword: {
+        const QString strUrl = resolveMultiplePlaceholdersRecursive(url(), maxDepth - 1);
+        return resolveUrlPlaceholder(strUrl, placeholderType);
+    }
+    case PlaceholderType::CustomAttribute: {
+        const QString key = placeholder.mid(3, placeholder.length() - 4); // {S:attr} => mid(3, len - 4)
+        return attributes()->hasKey(key) ? attributes()->value(key) : QString();
+    }
+    case PlaceholderType::Reference: {
+        // resolving references in format: {REF:<WantedField>@I:<uuid of referenced entry>}
+        // using format from http://keepass.info/help/base/fieldrefs.html at the time of writing,
+        // but supporting lookups of standard fields and references by UUID only
+
+        QString result;
+        QRegExp* referenceRegExp = m_attributes->referenceRegExp();
+        if (referenceRegExp->indexIn(placeholder) != -1) {
+            constexpr int wantedFieldIndex = 1;
+            constexpr int referencedUuidIndex = 2;
+            const Uuid referencedUuid(QByteArray::fromHex(referenceRegExp->cap(referencedUuidIndex).toLatin1()));
+            const Entry* refEntry = m_group->database()->resolveEntry(referencedUuid);
+            if (refEntry) {
+                const QString wantedField = referenceRegExp->cap(wantedFieldIndex).toLower();
+                if (wantedField == "t") result = refEntry->title();
```

Comment:
Don't put condition and body on the same line.

Reply:
Use brackets on all conditional statements (styleguide).

Reply:
Done